src/library/cmd_reader.h

```diff
@@ -184,5 +165,166 @@ namespace xlang::cmd
         }
 
         std::map<std::string_view, std::vector<std::string>> m_options;
+
+		template<typename O, typename L>
+		void extract_option(std::string const& sarg, O const& options, L &last) 
+		{
+			std::string_view svarg{ sarg };
+			if (svarg[0] == '-')
+			{
+				svarg.remove_prefix(1);
+				last = find(options, svarg);
+
+				if (last == options.end())
+				{
+					throw_invalid("Option '-", svarg, "' is not supported");
+				}
+
+				m_options.try_emplace(last->name);
+			}
+			else if (svarg[0] == '@')
+			{
+				svarg.remove_prefix(1);
+				extract_respose_file(svarg, options);
+			}
+			else if (last == options.end())
+			{
+				throw_invalid("Value '", svarg, "' is not supported");
+			}
+			else
+			{
+				m_options[last->name].push_back(std::string{ svarg });
+			}
+		}
+
+		template<typename O>
+		void extract_respose_file(std::string_view const& arg, O const& options)
+		{
+			path response_path{ std::string{ arg } };
+			std::wstring extension = response_path.extension();
+			std::transform(extension.begin(), extension.end(), extension.begin(), tolower);
+			if (is_directory(response_path) || extension == L".winmd")
+			{
+				throw_invalid("'@' is reserved for response files");
+			}
+			std::array<char, 8192> line_buf;
+			std::ifstream response_file(absolute(response_path));
+			while (response_file.getline(line_buf.data(), line_buf.size()))
+			{
+				size_t argc = 0;
+				std::vector<std::string> argv;
+
+				parse_command_line(line_buf.data(), argv, &argc);
+
+				auto last{ options.end() };
+				for (auto i = 0; i < argc; i++)
+				{
+					extract_option(argv[i], options, last);
+				}
+			}
+		}
+
+		template <typename Character>
+		static void parse_command_line(
```

Comment:
Where did you get this implementation?

Reply:
It's parse_command_line from argv_parsing.cpp, slightly modified. Sounds good.